exercises/giovanni_disiena/ex02_oo_basics/src/Rectangle.cpp

```diff
@@ -0,0 +1,48 @@
+#include "Rectangle.h"
+
+/** Constructs Rectangle object
+ *
+ * @param length: rectangle length
+ * @param width: rectangle width
+ *
+ */
+Rectangle::Rectangle(double length, double width)
+    : ShapeBase(4, "Rectangle"), m_length(length), m_width(width){};
+
+/** Constructs Rectangle (Square) object, setting width=length
+ *
+ * @param length: rectangle length
+ *
+ */
+Rectangle::Rectangle(double length)
+    : ShapeBase(4, "Square"), m_length(length), m_width(length) {}
```

Comment:
It would be better to treat square as a separate shape rather than as a rectangle with one length.